spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -131,34 +129,31 @@ public CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider(SQLAdmin sqlAdmin,
 	}
 
 	@Bean
-	@ConditionalOnMissingBean(DataSource.class)
-	public DataSource cloudSqlDataSource(CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider)
-			throws GeneralSecurityException, IOException, ClassNotFoundException {
-		HikariConfig hikariConfig = new HikariConfig();
-		hikariConfig.setDriverClassName(cloudSqlJdbcInfoProvider.getJdbcDriverClass());
-		hikariConfig.setJdbcUrl(cloudSqlJdbcInfoProvider.getJdbcUrl());
-		hikariConfig.setUsername(this.gcpCloudSqlProperties.getUserName());
-		hikariConfig.setPassword(this.gcpCloudSqlProperties.getPassword());
-		// Setting this is useful to disable connection initialization. Especially in unit tests.
-		hikariConfig.setInitializationFailFast(this.gcpCloudSqlProperties.isInitFailFast());
-
-		return new HikariDataSource(hikariConfig);
+	@Primary
+	public DataSourceProperties enhancedDatasourceProperties(DataSourceProperties properties,
```

Comment:
As part of this, we should probably remove `jdbcUrl`, `jdbcDriver`, `userName` and `password` from `GcpCloudSqlProperties`.
I think `DefaultCloudSqlJdbcInfoProvider` should also be modified to also take in a `DataSourceProperties` object and immediately return its `url` and `driverClassName` fields if they have a value (much like we do today if `GcpCloudSqlProperties.jdbcUrl` has a value).

Also, I wonder which `spring-cloud-gcp-starter-sql` dependency is instantiating the `DataSource` bean from `DataSourceProperties`? I wonder is it `spring-jdbc`?

Reply:
`GcpCloudSqlProperties` would only contain the GCP Cloud SQL specific properties, which is actually fair game. We can leave it to the `CloudSqlJdbcInfoProviders` to enforce the defaults, e.g., user="root", password="", jdbcDriver="com.mysql.jdbc.Driver", etc.